tests/cupy_tests/core_tests/test_raw.py

```diff
@@ -341,12 +342,20 @@ def use_temporary_cache_dir():
                 yield path
 
 
+_in_memory_cache = os.environ.get('CUPY_CACHE_IN_MEMORY')
+
+
 @testing.parameterize(*testing.product({
     'backend': ('nvrtc', 'nvcc'),
+    'in_memory': ('1', '0'),
 }))
 class TestRaw(unittest.TestCase):
 
     def setUp(self):
+        if self.backend == 'nvcc' and self.in_memory == '1':
+            self.skipTest('nvcc does not support in-memory cache')
+        os.environ['CUPY_CACHE_IN_MEMORY'] = self.in_memory
```

Comment:
Could you avoid mutating `os.environ`?

Reply:
I have a fresh thought this morning. Let me try if I can mock it with `functools.partial`. We need to mock the last argument in `_compile_with_cache_cuda(..., cache_in_memory)`.

Reply:
How about mocking `_get_bool_env_variable`?

Reply:
This sounds better than what I have! Let me give it a try...

Reply:
How do I mock `_get_bool_env_variable` such that when `name='CUPY_CACHE_IN_MEMORY'` we can change its `default` argument, but a call to `_get_bool_env_variable('CUPY_CACHE_IN_MEMORY', False)` is still valid?

Reply:
Take a look at 50b0430.